marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/flow/updateFlow.sjs

```diff
@@ -0,0 +1,71 @@
+const flowService = require("../lib/flowService.sjs");
+const hubTest = require("/test/data-hub-test-helper.xqy");
+const hubJsTest = require("/test/data-hub-test-helper.sjs");
+const stepService = require("../lib/stepService.sjs");
+const test = require("/test/test-helper.xqy");
+
+const assertions = [];
+const newFlowName = "newTestFlow";
+const stepList = {};
+
+let newFlow = {
+  "name": newFlowName,
+  "description": "description",
+  "steps": {}
+}
+
+// Create a flow and add steps
+let serviceResponse = flowService.createFlow(newFlowName, "description");
+["firstMapper", "secondMapper", "thirdMapper"].forEach(mappingName => {
+  let info = {
+    name: mappingName,
+    targetEntityType: "http://example.org/Customer-0.0.1/Customer",
+    selectedSource: "collection"
+  };
+  stepService.saveStep("mapping", info);
+  flowService.addStepToFlow(newFlowName, "mapping", mappingName);
+});
+
+// Get flow and verify it has 3 steps
+let flowTest = flowService.getFlow(newFlowName);
+assertions.push(
+  test.assertEqual(3, Object.keys(flowTest.steps).length),
+  test.assertEqual("firstMapper-mapping", flowTest.steps["1"].stepId),
+  test.assertEqual("secondMapper-mapping", flowTest.steps["2"].stepId),
+  test.assertEqual("thirdMapper-mapping", flowTest.steps["3"].stepId)
+);
+
+// Update the description and verify
+newFlow.description = "modified";
+serviceResponse = flowService.updateFlow(newFlow);
+assertions.push(
+  test.assertEqual(newFlow.description, serviceResponse.description)
```

Comment:
Can you add a couple of asserts to make sure that step order is unchanged ?

Reply:
Sure